components/amazon/amazon.go

```diff
@@ -48,35 +50,42 @@ func (amazon *CreateClusterAmazon) Validate() error {
 		amazon.Master.InstanceType = constants.AmazonDefaultMasterInstanceType
 	}
 
-	// ---- [ Node check ] ---- //
-	if amazon.Node == nil {
-		msg := "Required field 'node' is empty."
+	if len(amazon.NodePools) == 0 {
+		msg := "At least one 'nodePool' is required."
 		return errors.New(msg)
 	}
 
-	// ---- [ Node image check ] ---- //
-	if len(amazon.Node.Image) == 0 {
-		return errors.New("Required field 'image' is empty ")
-	}
-
-	// ---- [ Node min count check ] ---- //
-	if amazon.Node.MinCount == 0 {
-		amazon.Node.MinCount = constants.AmazonDefaultNodeMinCount
-	}
-
-	// ---- [ Node max count check ] ---- //
-	if amazon.Node.MaxCount == 0 {
-		amazon.Node.MaxCount = constants.AmazonDefaultNodeMaxCount
-	}
-
-	// ---- [ Node min count <= max count check ] ---- //
-	if amazon.Node.MaxCount < amazon.Node.MinCount {
-		return errors.New("maxCount must be greater than mintCount")
-	}
-
-	// ---- [ Node spot price ] ---- //
-	if len(amazon.Node.SpotPrice) == 0 {
-		amazon.Node.SpotPrice = constants.AmazonDefaultNodeSpotPrice
+	for _, amazonNode := range amazon.NodePools {
+
+		// ---- [ Node image check ] ---- //
+		if len(amazonNode.InstanceType) == 0 {
+			return errors.New("Required field 'instanceType' is empty ")
+		}
+
+		// ---- [ Node image check ] ---- //
+		if len(amazonNode.Image) == 0 {
+			return errors.New("Required field 'image' is empty ")
+		}
+
+		// ---- [ Node min count check ] ---- //
+		if amazonNode.MinCount == 0 {
+			amazonNode.MinCount = constants.AmazonDefaultNodeMinCount
+		}
+
+		// ---- [ Node max count check ] ---- //
+		if amazonNode.MaxCount == 0 {
+			amazonNode.MaxCount = constants.AmazonDefaultNodeMaxCount
+		}
+
+		// ---- [ Node min count <= max count check ] ---- //
+		if amazonNode.MaxCount < amazonNode.MinCount {
+			return errors.New("maxCount must be greater than mintCount")
```

Comment:
Let's introduce typed error (see for example `constants.ErrorNodePoolNotProvided`) instead of generic errors. This also will help with writing unit tests later to check that the appropriate errors are returned by functions.